### ISSUE TITLE
Handle updating a subDoc path when an array changes

### ISSUE THREAD

**Issue by salzhrani** [limeapps/ember-share#1]
Currently ShareProxy properties have a share subDocument under _context. each subDoc has a _path array which identifies its location within the share document. as explained [here](https://github.com/share/ShareJS/wiki/0.7-JSON-Client-API#a-note-on-paths). Currently when array operations happen the share document gets updated, while subDocs path do not causing the to have invalid indices. There is code in place to correct update a ShareArray own elements but it doesn't update elements further down the tree if any.
tl;dr 
need robust updating of paths.

**Comment by krisselden** [comment]
I thought Share.js subdocs were supposed to keep their path up to date, is this what https://github.com/share/ShareJS/blob/master/lib/types/json-api.js#L280 is supposed to do?

**Comment by salzhrani** [comment]
It is called only if there Is a move operation ( which we haven't
implemented yet ).

**Comment by krisselden** [comment]
@salzhrani have you tried opening an issue on share.js?

**Comment by salzhrani** [comment]
I did here
https://github.com/share/ShareJS/issues/296
But it didn't help

**Comment by krisselden** [comment]
So I think this is because Racer models an array of models as {model_ids: [1,2,3], models: {1: {}, 2: {}, 3:{}}} so that moving a model's order doesn't affect other editing context paths.
